src/frontend/app.ts

```diff
@@ -97,6 +97,17 @@ export default (octokitApp: App): Express => {
 
 	app.use(logMiddleware);
 
+	// Catch non successful responses
+	app.use((req, res, next) => {
+		res.once("finish", () => {
+			// if status is under 200 or higher/equal to 400, but not 503 in maintenance mode, log it to figure out issues
+			if ((res.statusCode < 200 || res.statusCode >= 400) && !(res.statusCode === 503 && booleanFlag(BooleanFlags.MAINTENANCE_MODE, false))) {
+				req.log.warn({ res, req }, `Returning HTTP response of '${res.statusCode}' for path '${req.path}'`);
```

Comment:
Maybe not in this PR, but later: we should probably expose this as a metric with the path and status as tags, so we can create a chart on top of this.